forms/listbox.cpp

```diff
@@ -285,7 +285,10 @@ void ListBox::clear()
 void ListBox::addItem(sp<Control> Item)
 {
 	Item->setParent(shared_from_this());
-	Item->ToolTipFont = this->ToolTipFont;
+	if (!Item->ToolTipFont && this->ToolTipFont)
```

Comment:
You could just simplify this to if (!Item->ToolTipFont), as if this->ToolTipFont is null then you're just setting null to null so no change

Reply:
But long term - we should probably cope with null fonts - probably implying some "default"?

Or make the Control constructor set a default, so it'll never be null (unless someone explicitly sets it)

Reply:
Actually I changed my mind: now I think it will be better if we avoid implicit inheritance for control attributes altogether. Sometimes listbox items are more complex than a single widget and inheriting attributes only to the top-level widget may be confusing and inheriting to the whole subtree may be too complex/expensive. I think it will be better to manually inherit each attribute in place where required.

I thought about the "default" font, but I don't recall that we have a fallback font defined in the framework? We don't want to tie the control library to a specific set of required resources either